### ISSUE TITLE
[Feature Request] More Flexible Image Gallery Widget(s)

### ISSUE THREAD

**Issue by schazzw** [pojome/elementor#3295]
Hey, I'm requesting that you either make the image gallery widget more flexible, or add additional ones that allow proper display of images of different dimensions (for instance including horizontal and vertical images in the same gallery). Here is an example of the issue. If you look at the link, the top image is a gallery from a weebly site I'm converting to wordpress/elementor. The bottom image is from the elementor image gallery widget, the same exact images with the same exact dimensions: https://www.screencast.com/t/QrAn4Hw2w

As you can see, the Weebly gallery widget has a setting to display only a part of the images that dont have the proper dimensions, and it looks fine. I was actually able to get a similar look in the elementor image gallery widget, but only if the images are set to 150x150 thumbnail, which is really small, and square, which isn't ideal. Alternatively, you could allow for a masonry display.

Here is a screenshot of the best I could do with it. The top gallery I left in 150x150, the bottom I left bigger, so the vertical images display entirely, but the spacing isn't the same between them as it is between other images, and I have to have them at the bottom for it to not look ridiculous: https://www.screencast.com/t/PFFTEN3t7Im

So basically, if you could give better options for displaying images of different dimensions in the same gallery, that would be great.

Thanks!

**Comment by bainternet** [comment]
Currently, this feature is not supported, but we will look into it in the future if we see a large demand for it.

Thanks for the feedback.
